Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM python:3.10.11
+
+# Install JupyterHub and dependencies
+RUN pip3 --disable-pip-version-check install --no-cache-dir \
+    jupyterhub==3.1.1 \
+    notebook==6.5.2 \
+    jupyterlab==3.5.3 \
+    jupyterlab-server==2.16.0
+
+# Install AlphaD3M and dependencies
+# RUN pip3 --disable-pip-version-check install --no-cache-dir \
+# 	alpha-automl
+
+ADD . /alpha-automl/
+WORKDIR /alpha-automl/
+ARG BUILD_OPTION='full'
+RUN pip3 install -e .[$BUILD_OPTION]
+
+# Create a user, since we don't want to run as root
+RUN useradd -m eden
+ENV HOME=/home/eden
+WORKDIR $HOME
+USER eden
+COPY --chown=eden examples /home/eden/examples
+
+ARG TOKEN=-1
+RUN mkdir -p $HOME/.jupyter/
+RUN if [ $TOKEN!=-1 ]; then echo "c.NotebookApp.token='$TOKEN'" >> $HOME/.jupyter/jupyter_notebook_config.py; fi
```

Comment:
Is this token configuration a requirement for JupyterHub? It looks like the token can also be controlled by the environment variable `JUPYTER_TOKEN` (https://jupyter-notebook.readthedocs.io/en/stable/config.html). If possible, using an environment variable may be better as it doesn't require rebuilding the image to change the token.

Reply:
This is used solely for docker build, the builder can add their own token, or simply just without it, like this:
`docker build --build-arg TOKEN='' --build-arg BUILD_OPTION='full' .`

Reply:
I get that. What I'm not sure of is if there are any advantages of hardcoding the token at build time (as opposed to providing the token at runtime using an environment variable). If you just remove those lines, we could run the image using the following command:

```
docker run -p 8888:8888 -e JUPYTER_TOKEN="<my-token>" ghcr.io/vida-nyu/alpha-automl
```